File: C++/Hard/Accepted/minimum-swaps-to-make-sequences-increasing_5.cpp
class Solution {
public:
    int minSwap(vector<int>& nums1, vector<int>& nums2) {
        
        int n = nums1.size();
        vector<int> keep(n, 0), swap(n, 0);
        keep[0] = 0;
        swap[0] = 1;
        for (int i = 1; i < n; i++) {
            keep[i] = swap[i] = INT_MAX;
            if (nums1[i] > nums1[i-1] && nums2[i] > nums2[i-1]) {
                keep[i] = keep[i-1];
                swap[i] = swap[i-1] + 1;
            }
            if (nums1[i] > nums2[i-1] && nums2[i] > nums1[i-1]) {
                keep[i] = min(keep[i], swap[i-1]);
                swap[i] = min(swap[i], keep[i-1] + 1);
            }
        }
        return min(keep[n-1], swap[n-1]);
    }
};

File: Rust/Hard/WrongAnswer/minimum-swaps-to-make-sequences-increasing_5.rs
impl Solution {
    pub fn min_swap(nums1: Vec<i32>, nums2: Vec<i32>) -> i32 {
        
        use std::cmp::min;
        let n = nums1.len();
        let mut dp1 = vec![0; n];
        let mut dp2 = vec![1; n];
        for i in 1..n {
            if nums1[i] > nums1[i-1] && nums2[i] > nums2[i-1] {
                if nums1[i] > nums2[i-1] && nums2[i] > nums1[i-1] {
                    dp2[i] = min(dp1[i-1], dp2[i-1]) + 1;
                }
                dp1[i] = dp1[i-1];
            } else {
                dp1[i] = dp2[i-1];
                dp2[i] = dp1[i-1] + 1;
            }
        }
        min(dp1[n-1], dp2[n-1])
    }
}